### PR TITLE
some small fixes from sycl branch

### PR DESCRIPTION
**Pull request by jcosborn**
fix some warnings from clang++
fix some headers needed in sycl branch

### CODE REVIEW

File: include/eigen_helper.h [hunk 1]
```diff
@@ -9,6 +9,7 @@
 #define EIGEN_DONT_VECTORIZE
 #endif
 
+#include <math.h>
```
Comment (maddyscientist) [inline include/eigen_helper.h]: Probably should be `cmath` here since we're a C++ library 😄

File: include/targets/generic/load_store.h [hunk 2]
```diff
@@ -1,4 +1,5 @@
 #pragma once
+#include <target_device.h>
```
Comment (maddyscientist) [inline include/targets/generic/load_store.h:3]: Super pedant comment here (sorry): prefer a line space between these two lines.

**Review by maddyscientist — APPROVED** [review]
Looks good.  Some a couple of trivial fixes, but we can merge once those are done.

File: include/eigen_helper.h (hunk 1 above)
Comment (jcosborn) [reply include/eigen_helper.h]: 062dc04e98a324dc503822ac992771308e9cb8ab

File: include/targets/generic/load_store.h (hunk 2 above)
Comment (jcosborn) [reply include/targets/generic/load_store.h:3]: 062dc04e98a324dc503822ac992771308e9cb8ab

**Review by maddyscientist — APPROVED** [review]

**Pull request merged**